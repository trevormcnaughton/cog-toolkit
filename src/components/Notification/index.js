// @flow
type Props = {
  text?: string,
  warning?: boolean,
  info?: boolean,
  className?: string,
  children?: any,
  error?: boolean,
}

import React from 'react'
import styled from 'styled-components'
import theme from '../../util/theme'

const Notification = ({
  children,
  className,
  error,
  info,
  style,
  text,
  warning,
}: Props) => {
  const Notification = styled.section`
    width: 100%;
    padding: 1rem;
    border-radius: 2px;
    background: #ffffff;
    border-left: 4px solid;
    border-color: ${props => {
        if (props.info) {
          return theme.color.blue
        } else if (props.warning) {
          return theme.color.warning
        } else if (props.error) {
          return theme.color.error
        } else {
          return theme.color.grey
        }
      }
    };
    box-shadow: 0 2px 8px 0 rgba(52,55,65,0.16);
  `

  return (
    <Notification
      info={info}
      warning={warning}
      error={error}
      style={style}>
      {children ? children : text}
    </Notification>
  )
}

export default Notification
