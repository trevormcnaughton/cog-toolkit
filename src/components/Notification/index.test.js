import React from 'react'
import renderer from 'react-test-renderer'

import Notification from './index'

describe('Notification', () => {
  describe('snapshots', () => {
    it('renders a default notification', () => {
      const tree = renderer.create(
        <Notification>Default Notification</Notification>
      )

      expect(tree).toMatchSnapshot()
    })
    it('renders a warning notification', () => {
      const tree = renderer.create(
        <Notification warning>Default Notification</Notification>
      )

      expect(tree).toMatchSnapshot()
    })
    it('renders an informational notification', () => {
      const tree = renderer.create(
        <Notification info>Default Notification</Notification>
      )

      expect(tree).toMatchSnapshot()
    })
    it('renders an error notification', () => {
      const tree = renderer.create(
        <Notification error>Default Notification</Notification>
      )

      expect(tree).toMatchSnapshot()
    })
  })
})
