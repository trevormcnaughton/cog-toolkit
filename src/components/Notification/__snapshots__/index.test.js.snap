// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification snapshots renders a default notification 1`] = `
<section
  className="sc-bdVaJa kvhSMU"
  style={undefined}
>
  Default Notification
</section>
`;

exports[`Notification snapshots renders a warning notification 1`] = `
<section
  className="sc-bwzfXH hkyWQW"
  style={undefined}
>
  Default Notification
</section>
`;

exports[`Notification snapshots renders an error notification 1`] = `
<section
  className="sc-bxivhb dpIgyD"
  style={undefined}
>
  Default Notification
</section>
`;

exports[`Notification snapshots renders an informational notification 1`] = `
<section
  className="sc-htpNat kVstz"
  style={undefined}
>
  Default Notification
</section>
`;
